api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java

```diff
@@ -0,0 +1,261 @@
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldCreateAbstractCloudRegionForAdmin() {
+        doReturn(region).when(mockCloudRegionManager).create(cloudRegionDTO);
+
+        assertThat(cloudRegionApiService.create(cloudRegionDTO)).isEqualTo(region);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    @WithMockUser(roles = SIMPLE_USER_ROLE)
+    public void shouldNotCreateAbstractCloudRegionForNotAdmin() {
+        doReturn(region).when(mockCloudRegionManager).create(cloudRegionDTO);
+
+        cloudRegionApiService.create(cloudRegionDTO);
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldUpdateAbstractCloudRegionForAdmin() {
+        doReturn(region).when(mockCloudRegionManager).update(region.getId(), cloudRegionDTO);
+
+        assertThat(cloudRegionApiService.update(region.getId(), cloudRegionDTO)).isEqualTo(region);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    @WithMockUser(roles = SIMPLE_USER_ROLE)
+    public void shouldNotUpdateAbstractCloudRegionForNotAdmin() {
+        doReturn(region).when(mockCloudRegionManager).update(region.getId(), cloudRegionDTO);
+
+        cloudRegionApiService.update(region.getId(), cloudRegionDTO);
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldDeleteAbstractCloudRegionForAdmin() {
+        doReturn(region).when(mockCloudRegionManager).delete(region.getId());
+
+        assertThat(cloudRegionApiService.delete(region.getId())).isEqualTo(region);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    @WithMockUser(roles = SIMPLE_USER_ROLE)
+    public void shouldDeleteAbstractCloudRegionForNotAdmin() {
+        doReturn(region).when(mockCloudRegionManager).delete(region.getId());
+
+        cloudRegionApiService.delete(region.getId());
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldLoadAllAvailableCloudsForAdmin() {
+        doReturn(availableCloudsList).when(mockCloudRegionManager).loadAllAvailable(CloudProvider.AWS);
+
+        assertThat(cloudRegionApiService.loadAllAvailable(CloudProvider.AWS)).isEqualTo(availableCloudsList);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    @WithMockUser(roles = SIMPLE_USER_ROLE)
+    public void shouldNotLoadAllAvailableCloudsForNotAdmin() {
+        doReturn(availableCloudsList).when(mockCloudRegionManager).loadAllAvailable(CloudProvider.AWS);
+
+        cloudRegionApiService.loadAllAvailable(CloudProvider.AWS);
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnAllCloudRegionsForAdmin() {
+        doReturn(singleRegionList).when(mockCloudRegionManager).loadAll();
+
+        final List<? extends AbstractCloudRegion> abstractCloudRegions = cloudRegionApiService.loadAll();
+        final AbstractCloudRegion resultRegion = abstractCloudRegions.get(0);
+
+        assertThat(abstractCloudRegions).isEqualTo(singleRegionList);
```

Comment:
The `singleRegionList` object was mutated during `@PostFilter` operation. This way this assertion is useless and we need another one. For example, let's check elements count (in this case we shall move `abstractCloudRegions.get(0)` after checking size).

Reply:
Done